source/Calamari/Integration/FileSystem/PackageStore.cs

```diff
@@ -108,57 +72,56 @@ public StoredPackage GetPackage(string prefix, PackageMetadata metadata)
 
         public IEnumerable<StoredPackage> GetNearestPackages(string packageId, SemanticVersion version, int take = 5)
         {
-            var root = GetPackageRoot(null);
-            fileSystem.EnsureDirectoryExists(root);
+            fileSystem.EnsureDirectoryExists(rootDirectory);
 
-            return (
-                from file in fileSystem.EnumerateFilesRecursively(root, packageId + "*.nupkg-*")
-                let package = ReadPackageFile(file)
+            var zipPackages =
+                from filePath in fileSystem.EnumerateFilesRecursively(rootDirectory, packageId + "*.nupkg-*")
+                let zip = ReadZipPackage(filePath)
+                where zip != null && zip.Id == packageId && zip.Version < version
```

Comment:
Should this be `< version` or `<= version`? Wouldn't an identical package version be *nearest* the one I want?